graph/src/schema/input_schema.rs

```diff
@@ -79,11 +81,13 @@ impl InputSchema {
                         .iter()
                         .map(|field| pool.lookup(&field.name).unwrap())
                         .collect();
-                    (EntityType::from(obj_type), fields)
-                }),
+                    EntityType::new(&pool, &obj_type.name).map(|t| (t, fields))
+                })
+                .collect::<Result<Vec<_>, _>>()
+                .unwrap(),
         );
 
-        let poi_type = EntityType::new(POI_OBJECT.to_string());
+        let poi_type = EntityType::new(&pool, POI_OBJECT).unwrap();
```

Comment:
maybe it could be better to just have some of these functions separately like `new_poi_object` which uses the const internally, leaks less details and could avoid the unwrap